Assets/Scripts/Effects/StunEffect.cs

```diff
@@ -2,27 +2,40 @@
 using UnityEngine;
 using UtilityExtensions;
 
+using EM = EventsManager;
 public class StunEffect : MonoBehaviour
 {
     public float flashInterval = 0.1f;
     private bool stopEffect = false;
 
     private void Start()
     {
-        PlayerStateManager stateManager = this.EnsureComponent<PlayerStateManager>();
+        // PlayerStateManager stateManager = this.EnsureComponent<PlayerStateManager>();
+        EM.onPlayerStunned += (EM.onPlayerStunnedArgs args) => {
+            Player player = args.player;
```

Comment:
Note: ignore this usage of events. This is not actually how I intend to use them -- I was just porting existing semantics to use EventManager syntax.

Specifically, I would _**not**_ want the PlayerStun component to take action based on events like `onPlayerStunned`. Instead, I would want any `PlayerStun` to be the only component who's supposed to fire `onPlayerStunned`, and `PlayerStun` would do any internal stun-related logic in a private member function.